packages/babel-types/src/clone/cloneNode.js

```diff
@@ -38,6 +38,10 @@ export default function cloneNode<T: Object>(node: T, deep: boolean = true): T {
   if (type === "Identifier") {
     newNode.name = node.name;
 
+    if (has(node, "optional") && typeof node.optional === "boolean") {
```

Comment:
Probably checking for boolean is not necessary

Reply:
Yeah I was going to leave this exact comment, but then I tought that it would just clean up cases like this:
```json5
{
  type: "Identifier",
  optional: null
}
```
Also, I don't think that it would jave a big runtime overhead given that it won't be looled up on the prototype (because of the "has" check).